New warnings when compiling

I get these warnings since very recently:
```
[264/265] Building CXX object CMakeFiles/aspect.dir/source/simulator/solver.cc.o
In file included from /home/bangerth/p/deal.II/1/projects/aspect/include/aspect/simulator.h:59:0,
                 from /home/bangerth/p/deal.II/1/projects/aspect/source/simulator/solver.cc:22:
/home/bangerth/p/deal.II/1/projects/aspect/include/aspect/boundary_composition/interface.h:235:44: warning: attribute ignored [-Wattributes]
         find_boundary_composition_model () const;
                                            ^~~~~
```

I'll take care of it after talking to Wanying.

--- a/include/aspect/boundary_composition/interface.h
+++ b/include/aspect/boundary_composition/interface.h
@@ -230,8 +230,8 @@ namespace aspect
 
 
         template <typename BoundaryCompositionType>
-        BoundaryCompositionType *
         DEAL_II_DEPRECATED
+        BoundaryCompositionType *
         find_boundary_composition_model () const;
 
         /**

